Sources/TuistCore/Graph/GraphTraverser.swift

```diff
@@ -244,8 +244,16 @@ public class GraphTraverser: GraphTraversing {
         return references
     }
 
-    // swiftlint:disable:next function_body_length
+    public func searchablePathDependencies(path: AbsolutePath, name: String) throws -> Set<GraphDependencyReference> {
+        try linkableDependencies(path: path, name: name, isForSearchPath: true)
+    }
+
     public func linkableDependencies(path: AbsolutePath, name: String) throws -> Set<GraphDependencyReference> {
+        try linkableDependencies(path: path, name: name, isForSearchPath: false)
+    }
+
+    // swiftlint:disable:next function_body_length
+    public func linkableDependencies(path: AbsolutePath, name: String, isForSearchPath: Bool) throws -> Set<GraphDependencyReference> {
```

Comment:
I recall you had some doubts about the approach with `isForSearchPath`, it's fine given what's is needed here. One suggestion would be make the boolean reflect the specific action being taken, e.g. `excludeHostAppDependencies`  _(not sure if it's any better 😅)_

```swift
public func linkableDependencies(path: AbsolutePath, name: String, excludeHostAppDependencies: Bool) throws -> Set<GraphDependencyReference>
```

Reply:
I will change it to `shouldExcludeHostAppDependencies` :)